File: Backend/schemas/github.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

# 前端轉換後格式（GET /github/issues 的 response_model）
class GitHubAuthor(BaseModel):
    username: Optional[str]
    avatar: Optional[str]

class GitHubIssue(BaseModel):
    id: int
    title: str
    state: str
    created_at: datetime
    updated_at: Optional[datetime]
    url: str
    isPR: bool
    author: Optional[GitHubAuthor]
    labels: Optional[List[str]]
    comments: Optional[int]

File: Frontend/types/github.ts
export interface GitHubAuthor {
    username?: string
    avatar?: string
  }
  
  export interface GitHubIssue {
    id: number
    title: string
    state: string
    created_at: string
    updated_at?: string
    url: string
    isPR: boolean
    author?: GitHubAuthor
    labels?: string[]
    comments?: number
  }
